src/Data/Nat/Binary/Subtraction.agda

```diff
@@ -0,0 +1,344 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Subtraction on Bin and some of its property proofs.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Nat.Binary.Subtraction where
+
+open import Algebra using (Op₂; Magma)
+open import Algebra.Consequences.Propositional using (comm+distrˡ⇒distrʳ)
+open import Data.Empty using (⊥)
+open import Data.Fin using (#_)
+open import Data.Nat as ℕ using (ℕ)
+open import Data.Nat.Binary.Base
+open import Data.Nat.Binary.Properties
+import Data.Nat.Properties as ℕₚ
+open import Data.Product using (_×_; _,_; proj₁; proj₂; ∃)
+open import Data.Sum using (inj₁; inj₂)
+open import Data.Vec using ([]; _∷_)
+open import Function using (_∘_; _$_)
+open import Level using (0ℓ)
+open import Relation.Binary using (Tri; tri<; tri≈; tri>; _Preserves_⟶_;
+                                                          _Preserves₂_⟶_⟶_)
+open import Relation.Binary.PropositionalEquality
+open import Relation.Nullary using (Dec; yes; no)
+open import Relation.Nullary.Negation using (contradiction)
+open import Algebra.Definitions {A = ℕᵇ} _≡_
+open import Algebra.Properties.CommutativeSemigroup +-commutativeSemigroup
+  using (xy∙z≈y∙xz; x∙yz≈y∙xz)
+open import Algebra.Solver.CommutativeMonoid +-0-commutativeMonoid
+
```

Comment:
Could you add a section heading named "Definition" here?

Reply:
Done